File: app/src/main/java/ru/melod1n/schedule/SettingsActivity.java
package ru.melod1n.schedule;

import android.os.Bundle;

import androidx.annotation.LayoutRes;

import ru.melod1n.schedule.common.AppGlobal;
import ru.melod1n.schedule.current.BaseActivity;
import ru.melod1n.schedule.fragment.SettingsFragment;
import ru.melod1n.schedule.util.Util;
import ru.melod1n.schedule.widget.Toolbar;

public class SettingsActivity extends BaseActivity {

    private SettingsFragment fragment;
    private int layoutId = R.xml.fragment_settings;

    private Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        toolbar = findViewById(R.id.toolbar);
        toolbar.setNavigationIcon(R.drawable.ic_arrow_backward);
        toolbar.setNavigationOnClickListener(v -> onBackPressed());

        layoutId = AppGlobal.preferences.getInt("settings_layout", R.xml.fragment_settings);

        Bundle arguments = new Bundle();
        arguments.putInt("layout_id", layoutId);

        fragment = new SettingsFragment();
        fragment.setArguments(arguments);

        getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, fragment, fragment.getClass().getSimpleName()).commit();
    }

    @Override
    protected void init() {
        super.init();

        if (!created) return;
        recreate();
    }

    @Override
    public void onBackPressed() {
        if (fragment == null || fragment.onBackPressed())
            super.onBackPressed();
    }

    @Override
    public void setTitle(CharSequence title) {
        if (toolbar != null)
            toolbar.setTitle(title);
    }

    @Override
    public void setTitle(int titleId) {
        setTitle(getString(titleId));
    }

    public void setFragmentElement(@LayoutRes int layoutId) {
        this.layoutId = layoutId;
        AppGlobal.preferences.edit().putInt("settings_layout", layoutId).apply();
    }
}

File: app/src/main/java/ru/melod1n/schedule/activity/SettingsActivity.kt
package ru.melod1n.schedule.activity

import android.os.Bundle
import androidx.annotation.LayoutRes
import kotlinx.android.synthetic.main.toolbar.*
import ru.melod1n.schedule.R
import ru.melod1n.schedule.common.AppGlobal
import ru.melod1n.schedule.base.BaseActivity
import ru.melod1n.schedule.fragment.SettingsFragment

class SettingsActivity : BaseActivity() {

    private lateinit var fragment: SettingsFragment
    private var layoutId = R.xml.fragment_settings

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        toolbar.setNavigationIcon(R.drawable.ic_arrow_backward)
        toolbar.setNavigationOnClickListener { onBackPressed() }

        layoutId = AppGlobal.preferences.getInt("settings_layout", R.xml.fragment_settings)

        val arguments = Bundle()
        arguments.putInt("layout_id", layoutId)

        fragment = SettingsFragment()
        fragment.arguments = arguments

        supportFragmentManager.beginTransaction().replace(R.id.fragmentContainer, fragment, fragment.javaClass.simpleName).commit()
    }

    override fun init() {
        super.init()
        if (!created) return
        recreate()
    }

    override fun onBackPressed() {
        if (fragment.onBackPressed()) super.onBackPressed()
    }

    override fun setTitle(title: CharSequence) {
        if (toolbar != null) toolbar!!.title = title
    }

    override fun setTitle(titleId: Int) {
        title = getString(titleId)
    }

    fun setFragmentElement(@LayoutRes layoutId: Int) {
        this.layoutId = layoutId
        AppGlobal.preferences.edit().putInt("settings_layout", layoutId).apply()
    }

    override fun onDestroy() {
        AppGlobal.preferences.edit().remove("settings_layout").apply()
        super.onDestroy()
    }
}